### PR TITLE
postgresql: postgresql database chain implementation plus example

### PR DESCRIPTION
**Pull request by Oloruntobi1**
Add feature: sql chain

-- postgresql database

### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [ ] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

File: tools/sqldatabase/postgresql/postgresql.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+// NewPostgreSQL creates a new PostgreSQL engine.
+// The dsn is the data source name.(e.g. root:password@tcp(localhost:3306)/test).
+func NewPostgreSQL(dsn string) (sqldatabase.Engine, error) { //nolint:ireturn
+	db, err := sql.Open(EngineName, dsn)
+	if err != nil {
+		return nil, err
+	}
+	db.SetMaxOpenConns(32) //nolint:gomnd
```
Comment (tmc) [inline tools/sqldatabase/postgresql/postgresql.go:31]: why 32? i'd prefer to omit this and consider including it as an option

File: tools/sqldatabase/postgresql/postgresql.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	return &PostgreSQL{
+		db: db,
+	}, nil
+}
+
+func (p PostgreSQL) Dialect() string {
+	return EngineName
+}
+
+func (p PostgreSQL) Query(ctx context.Context, query string, args ...any) ([]string, [][]string, error) {
```
Comment (tmc) [inline tools/sqldatabase/postgresql/postgresql.go:42]: these methods should be documented

**Review by tmc — COMMENTED** [review]
looks pretty good, a few small comments

File: tools/sqldatabase/postgresql/postgresql.go (hunk 1 above)
Comment (Oloruntobi1) [reply tools/sqldatabase/postgresql/postgresql.go:31]: okay i'll omit it then.

**Comment by Oloruntobi1** [comment]
PTAL @tmc

**Comment by tmc** [comment]
we need to make the example ci step smarter..

**Pull request merged**